tests/test_spotify.py

```diff
@@ -185,5 +185,51 @@ def test_that_song_function_returns_None_when_error(self):
         self.assertRaises(SpotifyNotRunning, x)
 
 
+class WebTests(unittest.TestCase):
+    """
+    Unit tests for Chrome (with extension) for spotify web player
+    """
+
+    def setup(self):
+        pass
+
+    @patch('SwSpotify.spotify.get_info_web')
+    def test_that_artist_function_calls_get_info(self, mock):
+        """
+        test that test artist function calls get_info_chrome function
+        """
+        x = artist()
+        self.assertTrue(mock.called)
+
+    @patch('SwSpotify.spotify.get_info_web')
+    def test_that_song_function_calls_get_info(self, mock):
+        """
+        test that test song function calls get_info_web function
+        """
+        x = song()
+        self.assertTrue(mock.called)
+
+    def test_that_artist_function_returns_None_when_error(self):
+        """
+        test that test artist function returns None when the get_info_web function will return an error
+        """
+        x = artist
+        self.assertRaises(SpotifyNotRunning, x)
+
+    def test_that_song_function_returns_None_when_error(self):
+        """
+        test that test song function returns None when the get_info_web function will return an error
+        """
+        x = song
+        self.assertRaises(SpotifyNotRunning, x)
+
+    @patch('SwSpotify.spotify.get_info_web')
+    def test_that_get_info_returns_data(self, mock):
+        """
+        test that get_info_web returns song, artist (or None, None) if Spotify not running
+        """
+        self.assertEqual(get_info_web(), (song, artist))
```

Comment:
When I said (song, artist) I meant for some song artist pair you should patch the function internally and test that it works, like here (https://github.com/SwagLyrics/SwSpotify/blob/8e53ac1edc53b8fe49e4e4b047c1cddadd181a5f/tests/test_spotify.py#L69)

Currently, since you're patching the function you want to test, it will return true to any output that you give to it

Reply:
right, that makes more seance

Reply:
Is this the kind of test that you are looking for? @aadibajpai.

```python
    @patch('spotify_web.run', return_value={"title": "Darkside", "artist": "Alan Walker"})
    def test_get_info_web(self, mock_spotify_web):
        """
        test that get_info_web works
        """
        x = get_info_web()
        self.assertEqual(x, ("Darkside", "Alan Walker"))
```

Reply:
Yep you're on the right track.